pyccel/ast/numpyext.py

```diff
@@ -1679,52 +1820,44 @@ def keep_dims(self):
         """
         return self._keep_dims
 
-class NumpyArraySize(PyccelInternalFunction):
+
+class NumpySize(PyccelInternalFunction):
     """
-    Class representing a call to the numpy size function which
-    returns the shape of an object in a given dimension
+    Represent a call to numpy.size in the user code.
+
+    This wrapper class represents a call to the Numpy size function, which
+    returns the total number of elements in a multidimensional array, or the
+    number of elements along a given dimension.
+
+    Objects of this class are never present in the Pyccel AST, because the
+    class constructor returns objects of type `PyccelArraySize`, `Literal`, or
+    `PyccelArrayShapeElement`.
 
     Parameters
-    ==========
-    arg   : PyccelAstNode
-            A PyccelAstNode of unknown shape
-    axis  : int
-            The dimension along which the size is
-            requested
+    ----------
+    a : PyccelAstNode
+        An array of unknown size.
+
+    axis : int, optional
+        The dimension along which the size is requested.
+
```

Comment:
Looking at the code it seems that `axis` can also be a `Variable` or a `LiteralInteger` (or maybe an expression?). Maybe the more general `PyccelAstNode` would be more accurate. `int` can be kept if it is used as a shortcut in code written by developers, but I am not sure that this is useful for `NumpySize`

Reply:
Fixed in c82512e